ampd/src/solana/msg_verifier.rs

```diff
@@ -1,31 +1,30 @@
 use axelar_wasm_std::voting::Vote;
-use gmp_gateway::events::{CallContract, GatewayEvent};
+use gmp_gateway::events::{ArchivedCallContract, ArchivedGatewayEvent, GatewayEvent};
+use solana_sdk::pubkey::Pubkey;
 use solana_transaction_status::{
     option_serializer::OptionSerializer, EncodedConfirmedTransactionWithStatusMeta,
 };
-use std::{borrow::Cow, sync::Arc};
+use std::sync::Arc;
+use std::str::FromStr;
 use tracing::error;
 
 use crate::handlers::solana_verify_msg::Message;
 
-impl PartialEq<&Message> for GatewayEvent<'_> {
+impl PartialEq<&Message> for &ArchivedGatewayEvent {
     fn eq(&self, msg: &&Message) -> bool {
         match self {
-            GatewayEvent::CallContract(Cow::Owned(CallContract {
+            ArchivedGatewayEvent::CallContract(ArchivedCallContract {
                 sender,
                 destination_chain,
                 destination_address,
                 payload: _,
                 payload_hash,
-            })) => {
-                let event_dest_addr = String::from_utf8(destination_address.to_owned());
-                let event_dest_chain = String::from_utf8(destination_chain.to_owned());
-
-                event_dest_addr.is_ok()
-                    && sender.to_string() == msg.source_address
-                    && event_dest_chain.is_ok()
-                    && event_dest_addr.unwrap() == msg.destination_address
-                    && msg.destination_chain == event_dest_chain.unwrap()
+            }) => {
+                let msg_sender = Pubkey::from_str(msg.source_address.as_str());
+                msg_sender.is_ok()
+                    && sender == &msg_sender.unwrap().to_bytes()
```

Comment:
let's get rid of the unwrap. If the source address cannot be parsed then it's no equal.
```rust
let Ok(msg_sender) = Pubkey::from_str(msg.source_address.as_str()) {
   return false;
};
...
```

Reply:
fixed at c136ce88ef89c9a649e35a2a61e0157996f468f1